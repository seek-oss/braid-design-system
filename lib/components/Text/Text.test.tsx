import React from 'react';
import { Text } from '..';
import { render } from 'react-testing-library';
import forEachTheme from '../../test/utils/forEachTheme';

describe('Text', () => {
  forEachTheme(({ ThemeProvider }) => {
    test('Default', () => {
      const { container } = render(
        <ThemeProvider>
          <Text>Children</Text>
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Size', () => {
      const { container } = render(
        <ThemeProvider>
          <Text size="large">Children</Text>
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Color', () => {
      const { container } = render(
        <ThemeProvider>
          <Text color="positive">Children</Text>
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Weight', () => {
      const { container } = render(
        <ThemeProvider>
          <Text weight="strong">Children</Text>
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Baseline disabled', () => {
      const { container } = render(
        <ThemeProvider>
          <Text baseline={false}>Children</Text>
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('List item', () => {
      const { container } = render(
        <ThemeProvider>
          <Text component="li">Children</Text>
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
