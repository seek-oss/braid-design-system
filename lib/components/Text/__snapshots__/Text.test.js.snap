// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Theme: jobStreet Baseline disabled 1`] = `
<div
  class="Text__block Box__root borderColor__default"
>
  <span
    class="Text__block color__neutral fontFamily__body fontSize__body fontWeight__regular"
    style="transform: none;"
  >
    Children
  </span>
</div>
`;

exports[`Text Theme: jobStreet Color 1`] = `
<div
  class="Text__block Box__root borderColor__default"
>
  <span
    class="Text__block color__positive fontFamily__body fontSize__body fontWeight__regular"
  >
    Children
  </span>
</div>
`;

exports[`Text Theme: jobStreet Default 1`] = `
<div
  class="Text__block Box__root borderColor__default"
>
  <span
    class="Text__block color__neutral fontFamily__body fontSize__body fontWeight__regular"
  >
    Children
  </span>
</div>
`;

exports[`Text Theme: jobStreet List item 1`] = `
<li
  class="Text__block Text__listItem Box__root borderColor__default"
>
  <span
    class="Text__block color__neutral fontFamily__body fontSize__body fontWeight__regular"
  >
    Children
  </span>
</li>
`;

exports[`Text Theme: jobStreet Size 1`] = `
<div
  class="Text__block Box__root borderColor__default"
>
  <span
    class="Text__block color__neutral fontFamily__body fontSize__large fontWeight__regular"
  >
    Children
  </span>
</div>
`;

exports[`Text Theme: jobStreet Weight 1`] = `
<div
  class="Text__block Box__root borderColor__default"
>
  <span
    class="Text__block color__neutral fontFamily__body fontSize__body fontWeight__strong"
  >
    Children
  </span>
</div>
`;

exports[`Text Theme: seekAnz Baseline disabled 1`] = `
<div
  class="Text__block Box__root borderColor__default"
>
  <span
    class="Text__block color__neutral fontFamily__body fontSize__body fontWeight__regular"
    style="transform: none;"
  >
    Children
  </span>
</div>
`;

exports[`Text Theme: seekAnz Color 1`] = `
<div
  class="Text__block Box__root borderColor__default"
>
  <span
    class="Text__block color__positive fontFamily__body fontSize__body fontWeight__regular"
  >
    Children
  </span>
</div>
`;

exports[`Text Theme: seekAnz Default 1`] = `
<div
  class="Text__block Box__root borderColor__default"
>
  <span
    class="Text__block color__neutral fontFamily__body fontSize__body fontWeight__regular"
  >
    Children
  </span>
</div>
`;

exports[`Text Theme: seekAnz List item 1`] = `
<li
  class="Text__block Text__listItem Box__root borderColor__default"
>
  <span
    class="Text__block color__neutral fontFamily__body fontSize__body fontWeight__regular"
  >
    Children
  </span>
</li>
`;

exports[`Text Theme: seekAnz Size 1`] = `
<div
  class="Text__block Box__root borderColor__default"
>
  <span
    class="Text__block color__neutral fontFamily__body fontSize__large fontWeight__regular"
  >
    Children
  </span>
</div>
`;

exports[`Text Theme: seekAnz Weight 1`] = `
<div
  class="Text__block Box__root borderColor__default"
>
  <span
    class="Text__block color__neutral fontFamily__body fontSize__body fontWeight__strong"
  >
    Children
  </span>
</div>
`;

exports[`Text Theme: seekAsia Baseline disabled 1`] = `
<div
  class="Text__block Box__root borderColor__default"
>
  <span
    class="Text__block color__neutral fontFamily__body fontSize__body fontWeight__regular"
    style="transform: none;"
  >
    Children
  </span>
</div>
`;

exports[`Text Theme: seekAsia Color 1`] = `
<div
  class="Text__block Box__root borderColor__default"
>
  <span
    class="Text__block color__positive fontFamily__body fontSize__body fontWeight__regular"
  >
    Children
  </span>
</div>
`;

exports[`Text Theme: seekAsia Default 1`] = `
<div
  class="Text__block Box__root borderColor__default"
>
  <span
    class="Text__block color__neutral fontFamily__body fontSize__body fontWeight__regular"
  >
    Children
  </span>
</div>
`;

exports[`Text Theme: seekAsia List item 1`] = `
<li
  class="Text__block Text__listItem Box__root borderColor__default"
>
  <span
    class="Text__block color__neutral fontFamily__body fontSize__body fontWeight__regular"
  >
    Children
  </span>
</li>
`;

exports[`Text Theme: seekAsia Size 1`] = `
<div
  class="Text__block Box__root borderColor__default"
>
  <span
    class="Text__block color__neutral fontFamily__body fontSize__large fontWeight__regular"
  >
    Children
  </span>
</div>
`;

exports[`Text Theme: seekAsia Weight 1`] = `
<div
  class="Text__block Box__root borderColor__default"
>
  <span
    class="Text__block color__neutral fontFamily__body fontSize__body fontWeight__strong"
  >
    Children
  </span>
</div>
`;

exports[`Text Theme: wireframe Baseline disabled 1`] = `
<div
  class="Text__block Box__root borderColor__default"
>
  <span
    class="Text__block color__neutral fontFamily__body fontSize__body fontWeight__regular"
    style="transform: none;"
  >
    Children
  </span>
</div>
`;

exports[`Text Theme: wireframe Color 1`] = `
<div
  class="Text__block Box__root borderColor__default"
>
  <span
    class="Text__block color__positive fontFamily__body fontSize__body fontWeight__regular"
  >
    Children
  </span>
</div>
`;

exports[`Text Theme: wireframe Default 1`] = `
<div
  class="Text__block Box__root borderColor__default"
>
  <span
    class="Text__block color__neutral fontFamily__body fontSize__body fontWeight__regular"
  >
    Children
  </span>
</div>
`;

exports[`Text Theme: wireframe List item 1`] = `
<li
  class="Text__block Text__listItem Box__root borderColor__default"
>
  <span
    class="Text__block color__neutral fontFamily__body fontSize__body fontWeight__regular"
  >
    Children
  </span>
</li>
`;

exports[`Text Theme: wireframe Size 1`] = `
<div
  class="Text__block Box__root borderColor__default"
>
  <span
    class="Text__block color__neutral fontFamily__body fontSize__large fontWeight__regular"
  >
    Children
  </span>
</div>
`;

exports[`Text Theme: wireframe Weight 1`] = `
<div
  class="Text__block Box__root borderColor__default"
>
  <span
    class="Text__block color__neutral fontFamily__body fontSize__body fontWeight__strong"
  >
    Children
  </span>
</div>
`;
